web/src/main/java/no/nav/modiapersonoversikt/service/enhetligkodeverk/KodeverkConfig.kt

```diff
@@ -1,31 +1,61 @@
 package no.nav.modiapersonoversikt.service.enhetligkodeverk
 
-enum class KodeverkConfig(private val kilde: EnhetligKodeverk.Kilde) {
-    ARKIVTEMA(FellesKodeverkKilde("Arkivtemaer")),
-    BEHANDLINGSTEMA(FellesKodeverkKilde("Behandlingstema")),
-    KJONN(FellesKodeverkKilde("Kjønnstyper")),
-    LAND(FellesKodeverkKilde("Landkoder")),
-    SPRAK(FellesKodeverkKilde("Språk")),
-    RETNINGSNUMRE(FellesKodeverkKilde("Retningsnumre")),
-    POSTNUMMER(FellesKodeverkKilde("Postnummer")),
-    PERSONSTATUSER(FellesKodeverkKilde("Personstatuser")),
-    SIVILSTAND(FellesKodeverkKilde("Sivilstander")),
-    DISKRESJONSKODER(FellesKodeverkKilde("Diskresjonskoder")),
-    VALUTA(FellesKodeverkKilde("Valutaer")),
-    TEMA(FellesKodeverkKilde("Tema")),
-    SF_TEMAGRUPPER(SfHenvendelseKodeverkKilde());
+import no.nav.modiapersonoversikt.service.enhetligkodeverk.kodeverkproviders.KodeverkProviders
+import no.nav.modiapersonoversikt.service.enhetligkodeverk.kodeverkproviders.oppgave.OppgaveKodeverk
 
-    fun hentKodeverk(providers: KodeverkProviders) = kilde.hentKodeverk(providers)
+object KodeverkConfig {
+    @JvmField
+    val ARKIVTEMA = FellesKodeverkKilde("Arkivtemaer")
+    val BEHANDLINGSTEMA = FellesKodeverkKilde("Behandlingstema")
+    val KJONN = FellesKodeverkKilde("Kjønnstyper")
+    val LAND = FellesKodeverkKilde("Landkoder")
+    val SPRAK = FellesKodeverkKilde("Språk")
+    val RETNINGSNUMRE = FellesKodeverkKilde("Retningsnumre")
+    val POSTNUMMER = FellesKodeverkKilde("Postnummer")
+    val PERSONSTATUSER = FellesKodeverkKilde("Personstatuser")
+    val SIVILSTAND = FellesKodeverkKilde("Sivilstander")
+    val DISKRESJONSKODER = FellesKodeverkKilde("Diskresjonskoder")
+    val VALUTA = FellesKodeverkKilde("Valutaer")
+    val TEMA = FellesKodeverkKilde("Tema")
+    val SF_TEMAGRUPPER = SfHenvendelseKodeverkKilde()
+    val OPPGAVE = OppgaveKodeverkKilde()
+
+    fun values(): List<EnhetligKodeverk.Kilde<*, *>> {
```

Comment:
[] For å slippe å liste opp alle verdiene her så kan man gjøre noe alà; 
```kotlin
abstract class ObjectEnum<T> {
    private val values = mutableListOf<T>()
    fun values(): List<T> = values
    
    protected fun add(value: T): T {
        values.add(value)
        return value
    }
}


object KodeverkConfig : ObjectEnum<EnhetligKodeverk.Kilde<*, *>>() {
  @JvmField
  val ARKIVTEMA = add(FellesKodeverkKilde("Arkivtemaer"))
  val BEHANDLINGSTEMA = add(FellesKodeverkKilde("Behandlingstema"))
}
```
Det er da kallet til `add()` som legges det til i listen som returneres ved kall til `values()`

Reply:
Lurt!